3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +22,94 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+const USER_INTERFACE_ID = 'user-interface';
+const DOM_IMG_ID = 'image-api';
+const UI_MESSAGE_ID = 'ui-message';
+const API_URL = 'https://pokeapi.co/api/v2/pokemon/';
+const SELECT_ELEMENT_ID = 'select-pokes';
+
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    //Get data from response if it's ok.
+    if (response.ok) {
+      const data = await response.json();
+      return Promise.resolve(data);
+    }
+    console.log('Failed response status: ', response.status);
+  } catch (error) {
+    console.log(error.message);
+  }
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+//Boolean to check if img and select elements are already created on the DOM
+let isUIPopulated = false;
+
+//This is called with "get pokemons" button click
+async function fetchAndPopulatePokemons() {
+  //Get elements id if they're already created, else: create them.
+  const [ui, selectElement] = isUIPopulated
+    ? [
+        document.getElementById(USER_INTERFACE_ID),
+        document.getElementById(SELECT_ELEMENT_ID),
+      ]
+    : [createUIElements(), createElement('select', SELECT_ELEMENT_ID)];
+
+  //Fetch data and create an option element for each pokemon then add it to select.
+  fetchData(API_URL).then((data) => {
+    for (const pokemon of data.results) {
+      const optionElm = createElement('option', '', pokemon.name, pokemon.url);
+      selectElement.appendChild(optionElm);
+    }
+  });
+  ui.appendChild(selectElement);
+  //Append ui to body here to condition it on click
+  document.body.appendChild(ui);
+
+  selectElement.addEventListener('change', fetchImage);
+}
+
+//Called when select element is changed.
+function fetchImage() {
+  //Image element is already created in another method.
+  const img = document.getElementById(DOM_IMG_ID);
+  const optionElement = this.selectedOptions[0];
```

Comment:
No need to use `this` here. The best practice is to use the event object passed to the event listener:

```suggestion
function fetchImage(e) {
  //Image element is already created in another method.
  const img = document.getElementById(DOM_IMG_ID);
  const imgUrl = e.target.value;
```